devices/src/virtio/net.rs

```diff
@@ -80,6 +80,8 @@ struct NetEpollHandler {
     interrupt_status: Arc<AtomicUsize>,
     interrupt_evt: EventFd,
     rx_buf: [u8; MAX_BUFFER_SIZE],
+    tx_frame: [u8; MAX_BUFFER_SIZE],
+    tx_used_desc_heads: [u16; QUEUE_SIZE as usize],
```

Comment:
Maybe a bit late, but instead of polluting global state with local state, a better option would have been to use a MaybeUninit array and get a slice over it after initialization.

Reply:
MaybeUninit didn't exist at that point in time. So I say that you are a bit early, not a bit late.

Reply:
Thanks for the `MaybeUninit` array suggestion, it looks interesting.

How is the current version of the code polluting global state? It's the device state which at any one time may have a buffered/pending frame.

Reply:
To me it seems that this array is only using during process_tx and the data inside the array is only being using during a single call of process_tx, another call will not reuse this state. I haven't checked all the commits in this series though.